File: C#/0113. Path Sum II.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<IList<int>> PathSum(TreeNode root, int sum) {
        var result=new List<IList<int>>();
        if(root==null)
            return result;
        var rootQueue=new Queue<TreeNode>();
        rootQueue.Enqueue(root);
        var rootValQueue=new Queue<List<int>>();
        rootValQueue.Enqueue(new List<int>());
        var rootValSumQueue=new Queue<int>();
        rootValSumQueue.Enqueue(0);
        
        while(rootQueue.Count()!=0){
            TreeNode curRoot=rootQueue.Dequeue();
            List<int> curRootValList=rootValQueue.Dequeue();
            int curSum=rootValSumQueue.Dequeue();
            
            curRootValList.Add(curRoot.val);
            curSum+=curRoot.val;
            
            if(curRoot.left==null && curRoot.right==null && curSum==sum){
                result.Add(curRootValList);
            }
            else{
                if(curRoot.left!=null){
                    rootQueue.Enqueue(curRoot.left);
                    var tmp=new List<int>(curRootValList);
                    rootValQueue.Enqueue(tmp);
                    rootValSumQueue.Enqueue(curSum);
                }
                if(curRoot.right!=null){
                    rootQueue.Enqueue(curRoot.right);
                    var tmp=new List<int>(curRootValList);
                    rootValQueue.Enqueue(tmp);
                    rootValSumQueue.Enqueue(curSum);
                }
            }
        }
        return result;
    }
}

File: 20190519python/0113. Path Sum II.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def pathSum(self, root: TreeNode, sum: int) -> List[List[int]]:
        if not root:
            return []
        rep=[]
        self.getPathSum(root,sum,[],rep)
        return rep
    
    def getPathSum(self,root,target,res,rep):
        if not root.left and not root.right:
            if target==root.val:
                rep.append(res+[root.val])
        elif root.left and not root.right:
            self.getPathSum(root.left,target-root.val,res+[root.val],rep)
        elif not root.left and root.right:        
            self.getPathSum(root.right,target-root.val,res+[root.val],rep)
        else:
            self.getPathSum(root.left,target-root.val,res+[root.val],rep)
            self.getPathSum(root.right,target-root.val,res+[root.val],rep)
